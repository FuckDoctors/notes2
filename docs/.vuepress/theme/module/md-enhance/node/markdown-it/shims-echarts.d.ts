declare module '@MdECharts' {
  import type { ComponentOptions } from 'vue'

  const component: ComponentOptions

  export default component
}
