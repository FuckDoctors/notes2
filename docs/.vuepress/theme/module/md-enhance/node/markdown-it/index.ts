export * from './echarts'
export * from './playground'
