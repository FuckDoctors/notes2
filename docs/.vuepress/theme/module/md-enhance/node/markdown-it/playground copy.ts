import { hash } from '@vuepress/utils'
import type Token from 'markdown-it/lib/token'

const extensions = ['js', 'ts', 'vue', 'jsx', 'tsx', 'json']

export const playgroundRender = (tokens: Token[], index: number): string => {
  const { nesting, info } = tokens[index]
  const title = /^ *playground\s*(.*)\s*$/u.exec(info)
  const key = `playground-${hash(index)}`

  if (nesting === -1) return `</Playground>`

  const codeConfigs = {}

  let configKey
  for (let i = index; i < tokens.length; i++) {
    // console.log(i, tokens[i])
    const { type, content, info } = tokens[i]

    if (type === 'container_playground_close') break
    if (!content) continue

    if (type === 'inline') {
      const fileTitleReg = /^\s*::: *file\s*(.*)\s*$/u.exec(content)
      // const fileTitleReg = /^ *(?:file|imports)\s*(.*)\s*$/u.exec(info)
      const isImports = /^\s*::: *imports\s*$/u.test(content)

      if (!fileTitleReg && !isImports) {
        continue
      }

      if (isImports) {
        configKey = 'imports'
      } else {
        configKey = fileTitleReg[1]
      }
    }

    if (type === 'fence' && extensions.includes(info) && configKey) {
      codeConfigs[configKey] = {
        lang: info,
        content: content,
      }
    }
  }

  const config = encodeURIComponent(JSON.stringify(codeConfigs))

  return `<Playground title="${
    title?.[1] || ''
  }" id="${key}" config="${config}">`
}
