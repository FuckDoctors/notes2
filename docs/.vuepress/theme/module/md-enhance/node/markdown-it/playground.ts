import { hash } from '@vuepress/utils'
import type Token from 'markdown-it/lib/token'

const extensions = ['js', 'ts', 'vue', 'jsx', 'tsx', 'json']
const importKey = 'import-map.json'

export const playgroundRender = (tokens: Token[], index: number): string => {
  const { nesting, info } = tokens[index]
  const title = /^ *playground\s*(.*)\s*$/u.exec(info)
  const key = `playground-${hash(index)}`

  if (nesting === -1) return `</Playground>`

  const codeConfigs = {}
  let settings: string = null

  let configKey
  let isSettings
  for (let i = index; i < tokens.length; i++) {
    // console.log(i, tokens[i])
    const { type, content, info } = tokens[i]

    if (type === 'container_playground_close') break

    if (type === 'container_code-group-item_open') {
      const fileTitleReg = /^ *code-group-item\s*(.*)\s*$/u.exec(info)
      if (!fileTitleReg) {
        continue
      }
      configKey = fileTitleReg[1]
    } else if (type === 'inline') {
      configKey = null
      // const isImports = /^\s*::: *imports\s*$/u.test(content)
      const imports = /^\s*::: *imports\s*(.*)\s*$/u.exec(content)

      isSettings = /^\s*::: *settings\s*$/u.test(content)

      if (imports) {
        const importMap = imports[1]
        if (importMap.length > 1) {
          configKey = imports[1]
        } else {
          configKey = importKey
        }
      } else if (isSettings) {
        // ...
      } else {
        continue
      }
    }

    if (!content) continue

    if (type === 'fence' && extensions.includes(info) && configKey) {
      if (importKey === configKey) {
        codeConfigs[configKey] = {
          lang: info,
          content: `{\n  "imports": ${content}  }`,
        }
      } else {
        codeConfigs[configKey] = {
          lang: info,
          content: content,
        }
      }
    }

    if (type === 'fence' && info === 'json' && isSettings) {
      settings = content.replace(/^\s+|\s+$/g, '').replace(/\/+$/, '')
    }
  }

  const config = encodeURIComponent(JSON.stringify(codeConfigs))
  const settingString = settings
    ? encodeURIComponent(settings)
    : encodeURIComponent('{}')

  return `<Playground title="${title?.[1] || ''}"
  id="${key}"
  settings="${settingString}"
  config="${config}">`
}
