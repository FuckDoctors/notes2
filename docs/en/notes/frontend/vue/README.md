---
index: true
icon: vue
category:
  - notes
  - frontend
  - vue
tag:
  - vue
---

# Vue Notes
