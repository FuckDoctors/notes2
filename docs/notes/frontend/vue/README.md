---
index: true
sidebarText: 概述
icon: vue
category:
  - 笔记
  - frontend
  - vue
tag:
  - vue
---

# Vue 笔记

## 概述

Vue 学习笔记

- [Vue2 学习笔记](https://github.com/FuckDoctors/hello-vue-2.0)

  只是照着官方指南敲了敲代码

- Vue3 学习笔记

  只是照着官方指南敲了敲代码

  - [第一部分](https://github.com/FuckDoctors/hello-vue3)
  - [第二部分](./vue3/README.md)

- [Vue.js 设计与实现](./vue3-book/)

  本书源代码: <https://github.com/HcySunYang/code-for-vue-3-book>

- [Pinia 学习笔记](./pinia/)

  只是照着官方指南敲了敲代码

<!-- more -->
