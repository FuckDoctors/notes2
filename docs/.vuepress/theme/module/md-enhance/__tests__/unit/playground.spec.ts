import { describe, expect, it } from 'vitest'
import MarkdownIt from 'markdown-it'

import { playground } from '../../node/markdown-it'

describe('playground', () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(playground)

  it('Should resolve playground info', () => {
    expect(
      markdownIt.render(
        `
::: playground-zhaobc Playground demo
@file App.vue
\`\`\`vue
<script setup>
import { ref } from 'vue'
const msg = ref('Hello World!')
</script>
<template>
  <h1>{{ msg }}</h1>
  <input v-model="msg" />
</template>
\`\`\`
@file Comp.vue
\`\`\`vue
<template>
  <div>Comp</div>
</template>
\`\`\`
@imports
\`\`\`json
{
  "imports": {
    "vue": "https://sfc.vuejs.org/vue.runtime.esm-browser.js"
  }
}
\`\`\`
:::
`,
        {}
      )
    ).toMatchSnapshot()
  })

  it('Should resolve playground info with settings', () => {
    expect(
      markdownIt.render(
        `
::: playground-zhaobc Playground demo2
@file App.vue
\`\`\`vue
<script setup>
import { ref } from 'vue'
const msg = ref('Hello World!')
</script>
<template>
  <h1>{{ msg }}</h1>
  <input v-model="msg" />
</template>
\`\`\`
@file Comp.vue
\`\`\`vue
<template>
  <div>Comp</div>
</template>
\`\`\`
@settings
\`\`\`json
{
  "mode": "external",
  "external": {
    "base": "https://element-plus.run/"
  }
}
\`\`\`
:::
`,
        {}
      )
    ).toMatchSnapshot()
  })

  it('Should resolve playground info with settings', () => {
    expect(
      markdownIt.render(
        `
::: playground-zhaobc#customId Playground demo2
@file App.vue
\`\`\`vue
<script setup>
import { ref } from 'vue'
const msg = ref('Hello World!')
</script>
<template>
  <h1>{{ msg }}</h1>
  <input v-model="msg" />
</template>
\`\`\`
@imports user-imports.json
\`\`\`json
{
  "imports": {
    "vue": "https://sfc.vuejs.org/vue.runtime.esm-browser.js"
  }
}
\`\`\`
@settings
\`\`\`json
{
  "mode": "external",
  "external": {
    "base": "https://element-plus.run/"
  }
}
\`\`\`
:::
`,
        {}
      )
    ).toMatchSnapshot()
  })
})
