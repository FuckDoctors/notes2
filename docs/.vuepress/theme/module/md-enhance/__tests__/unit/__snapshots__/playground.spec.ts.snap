// Vitest Snapshot v1

exports[`playground > Should resolve playground info 1`] = `
"<Playground title=\\"playground demo\\"
  id=\\"playground-1a96284a\\"
  settings=\\"%7B%7D\\"
  config=\\"%7B%22user-imports.json%22%3A%7B%22lang%22%3A%22json%22%2C%22content%22%3A%22%7B%5Cn%20%20%5C%22imports%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22vue%5C%22%3A%20%5C%22https%3A%2F%2Fsfc.vuejs.org%2Fvue.runtime.esm-browser.js%5C%22%5Cn%20%20%7D%5Cn%7D%5Cn%22%7D%7D\\"><p>:::: code-group</p>
<p>::: code-group-item App.vue</p>
<pre><code class=\\"language-vue\\">&lt;script setup&gt;
import { ref } from 'vue'

const msg = ref('Hello World!')
&lt;/script&gt;

&lt;template&gt;
  &lt;h1&gt;{{ msg }}&lt;/h1&gt;
  &lt;input v-model=&quot;msg&quot; /&gt;
&lt;/template&gt;
</code></pre>
<p>:::</p>
<p>::: code-group-item Comp.vue</p>
<pre><code class=\\"language-vue\\">&lt;template&gt;
  &lt;div&gt;Comp&lt;/div&gt;
&lt;/template&gt;
</code></pre>
<p>:::</p>
<p>::: imports</p>
<pre><code class=\\"language-json\\">{
  &quot;imports&quot;: {
    &quot;vue&quot;: &quot;https://sfc.vuejs.org/vue.runtime.esm-browser.js&quot;
  }
}
</code></pre>
<p>:::</p>
<p>::::</p>
</Playground>"
`;

exports[`playground > Should resolve playground info with settings 1`] = `
"<Playground title=\\"playground demo2\\"
  id=\\"playground-1a96284a\\"
  settings=\\"%7B%0A%20%20%22base%22%3A%20%22https%3A%2F%2Felement-plus.run%2F%22%0A%7D\\"
  config=\\"%7B%7D\\"><p>:::: code-group</p>
<p>::: code-group-item App.vue</p>
<pre><code class=\\"language-vue\\">&lt;script setup&gt;
import { ref } from 'vue'

const msg = ref('Hello World!')
&lt;/script&gt;

&lt;template&gt;
  &lt;h1&gt;{{ msg }}&lt;/h1&gt;
  &lt;input v-model=&quot;msg&quot; /&gt;
&lt;/template&gt;
</code></pre>
<p>:::</p>
<p>::: code-group-item Comp.vue</p>
<pre><code class=\\"language-vue\\">&lt;template&gt;
  &lt;div&gt;Comp&lt;/div&gt;
&lt;/template&gt;
</code></pre>
<p>:::</p>
<p>::::</p>
<p>::: settings</p>
<pre><code class=\\"language-json\\">{
  &quot;base&quot;: &quot;https://element-plus.run/&quot;
}
</code></pre>
<p>:::</p>
</Playground>"
`;
